luma/oled/device/__init__.py

```diff
@@ -743,3 +746,232 @@ def _set_position(self, top, right, bottom, left):
+            my_display.text = 'WS0010 Display\\nFont FT01 5x8'
+
+        For more details on how to use the 'text' interface see
+        :class:`luma.core.virtual.character`
+
+    .. versionadded:: 3.6.0
+    """
+    def __init__(self, serial_interface=None, width=100, height=16, undefined='_', font=None, selected_font=0, rotate=0, framebuffer="diff_to_previous", const=luma.oled.const.ws0010, **kwargs):
+        super(ws0010, self).__init__(const, serial_interface, **kwargs)
+        self.capabilities(width, height, rotate)
+        self.framebuffer = getattr(luma.core.framebuffer, framebuffer)(self)
+        self.font = font if font is not None else embedded_fonts(self._const.FONTDATA, selected_font=selected_font)
+        self._undefined = undefined
+        self.device = self
+
+        # Supported modes
+        supported = (width, height) in [(40, 8), (40, 16), (60, 8), (60, 16), (80, 8), (80, 16), (100, 8), (100, 16)]
+        if not supported:
+            raise luma.core.error.DeviceDisplayModeError(
+                "Unsupported display mode: {0} x {1}".format(width, height))
+
+        # In case display just powered up, sleep to be sure it has finished
+        # its internal initialization
+        sleep(0.5)
+        self._reset()
+        self.text = ""
+
+    def _reset(self):
+        """
+        WS0010 Initialization Routine
+
+        Reset (not really needed after power-on but does no harm and is useful
+        in case the display is already initialized)
+
+        Send five 0s in four bit mode (e.g. d4-d7 are the only pins that matter)
+        Set interface data length FUNCTIONSET|DL8 or FUNCTIONSET|DL4 (again in 4 bit mode)
+        Turn Display Off
+        Turn internal power off
+        Configure Entry Mode (Set Cursor to Right and Shift Off)
+        Set Graphics Mode and Internal Power On
+        Turn Display On (with Cursor Off and Blink Off)
+
+        Device is now ready to receive data
+        """
+
+        dl = 0x03 if self._bitmode == 8 else 0x02
+        self.command(0x00, 0x00, dl, self._const.FUNCTIONSET | (dl << 4))
+        self.command(self._const.DISPLAYOFF)  # Set Display Off
+        self.command(self._const.POWEROFF)
+        self.command(self._const.ENTRY)  # Set entry mode to direction right, no shift
+        self.command(self._const.POWERON | self._const.GRAPHIC)  # Turn internal power on and set into graphics mode
+        self.command(self._const.DISPLAYON)  # Turn Display back on
+
+    def display(self, image):
+        """
+        Takes a 1-bit :py:mod:`PIL.Image` and dumps it to the ws0010
+        OLED display.
+        """
+        assert(image.mode == self.mode)
+        assert(image.size == self.size)
+
+        image = self.preprocess(image)
+
+        if self.framebuffer.redraw_required(image):
+            # Expand bounding box to align to cell height boundary (8)
+            # Should consider whether this should be moved into framebuffer class
```

Comment:
Prefix comments like these with either TODO or FIXME so that the ide highlights them for future attention

Reply:
Added TODO to the comment.  I think it is probably ok as-is though so let me know if you want me to open an issue to address it, or remove the comment.  It was really just me capturing my thought process as I was coding the routine.